flaml/tune/tune.py

```diff
@@ -605,6 +612,136 @@ def easy_objective(config):
             _running_trial = old_running_trial
             _training_iteration = old_training_iteration
 
+    if use_spark:
+        # parallel run with spark
+        from flaml.utils import check_spark
+
+        check_spark()
+        try:
+            import pyspark
+            from pyspark.sql import SparkSession
+            from joblib import Parallel, delayed, parallel_backend
+            from joblibspark import register_spark
+        except ImportError as e:
+            raise ImportError(
+                f"{e}. Try pip install flaml[spark] or set use_spark=False."
+            )
+        from flaml.tune.searcher.suggestion import ConcurrencyLimiter
+        from .trial_runner import SparkTrialRunner
+
+        register_spark()
+        spark = SparkSession.builder.getOrCreate()
+        sc = spark._jsc.sc()
+        num_executors = (
+            len([executor.host() for executor in sc.statusTracker().getExecutorInfos()])
+            - 1
+        )
+        """
+        By default, the number of executors is the number of VMs in the cluster. And we can

```

Comment:
This chunk of docstr and the setting in line 662 of this file are quite informative and important. Should we make it more visible? For example, adding instructions or a reference in the automl.py file when [Spark is enabled](https://github.com/microsoft/FLAML/blob/62cd35322ecd90eef9d81f6318d0aa53512e652d/flaml/automl/automl.py#L688)

Reply:
Thanks for the suggestion. Docs updated.